src/app/game-state/home.service.ts

```diff
@@ -717,23 +750,33 @@ export class HomeService {
         }
       }
     }
+
     // if there's no autohome, autohome is finished, or there's enough money to cover buying a new plot of land and the home, try buying land.
     if ((!this.autoBuyHomeUnlocked ||
         this.homeValue >= this.autoBuyHomeLimit ||
         (this.characterService.characterState.money >= this.nextHome.cost + this.landPrice + priceBuffer )) &&
         this.autoBuyLandUnlocked){
-      //keep checking if we have the money for the land plus food and rent.
-      while ((this.land + this.fields.length + this.extraFields) < this.autoBuyLandLimit &&
-             (this.characterService.characterState.money >= this.landPrice + priceBuffer) ){
-        //break if reduced to money for home.
+      //check if we have the money for the land plus food and rent.
```

Comment:
Didn't we already do this code?

Reply:
This is part of removing the loop, though I'm not 100% on the logic here because it gets hectic, it should be good.

Reply:
Because we still want to check to see if they're building or saving for a home before we calculate the money that's available to use for automatic land purchase.

Reply:
You sound a little hesitant. Have you tested this?

Reply:
Tested it. Works exactly as I expect.